eclipse-theia/theia
# plugin: fix 'canSelectMany' implementation for dialogs

#### What it does


Fixes: #9277 

The following pull-request fixes the `dialogs-main` implementation for opening dialogs from the plugin-system when using the `canSelectMany` prop. Previously, the handling would fail when multiple `uris` are returned by the service. The commit updates the logic to handle the different return types from the dialog service.

#### How to test


1. start the application with the following [dialog-plugin](https://github.com/vince-fugnitto/vscode-dialog/releases/download/0.0.2/vscode-dialog-0.0.1.vsix).
2. ensure that `dialog: open` works (only a single uri can be selected).
3. ensure that `dialog: open many` works (multiple uris can be selected, and no error).
4. ensure that `dialog: save` still works correctly.
5. verify the changes on the `browser` and `electron` targets.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
